src/sentry/api/helpers/source_map_helper.py

```diff
@@ -267,21 +309,21 @@ def _get_releasefiles(release, organization_id):
     return data_sources
 
 
-def _find_url_prefix(filepath, artifact_name):
+def _find_url_prefix(filepath_str: str, artifact_name_str: str) -> str | None:
```

Comment:
beyond scope of this PR, but this could be implemented much cleaner using [pathlib](https://python.readthedocs.io/en/stable/library/pathlib.html) objects.

Reply:
I don't think pathlib adds anything here and in general makes things slower and introduces a coloring problem